### PR TITLE
Add clusterclass-related documentation

### LINKED ISSUE #2741: Add usage and developer documentation regarding ClusterClass-based cluster LCM.
**Bug description**

There should be documentation to support users and developers of Tanzu CLI regarding ClusterClass support.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [x] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [x] User Experience

**Expected behavior**

There should be a reliable location in the repository's docs/ directory to collect documentation pertaining to ClusterClass support 

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by vuil**
Initial drop of some documentation and developer notes regarding support
for ClusterClass-based cluster creation. More will be added over time as
necessary.

Signed-off-by: Vui Lam <vui@vmware.com>

### Which issue(s) this PR fixes

Partially-Fixes: #2741

### Describe testing done for PR

N/A

### Release note

```release-note
None
```

### PR Checklist

- x Squash the commits into one or a small number of logical commits
      
- x Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- x Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: docs/cli/tkgctl/management-cluster-using-clusterclass.md [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 
-Below is the sequence of management-cluster creation using clusterclass.
+Below is the sequence of steps involved in creating a management cluster ClusterClass.
 
 1. Deploy a bootstrap kind cluster
 1. Deploy `kapp-controller` on bootstrap cluster
 1. Deploy `providers` on bootstrap cluster
-1. Deploy `TKG meta package` to bootstrap cluster
+1. Deploy `tanzu-framework package` to bootstrap cluster
+1. Deploy `tkg composite package` to bootstrap cluster
```
Comment (randomvariable) [inline docs/cli/tkgctl/management-cluster-using-clusterclass.md:9]: I would like to see descriptions of these packages.

Comment (vuil) [reply docs/cli/tkgctl/management-cluster-using-clusterclass.md:9]: Ack, will provide some more information on those.

File: docs/cli/clusterclass/cluster-class-support.md [hunk 2]
```diff
@@ -9,35 +9,56 @@ scenarios at the server end in a reusable fashion, we will also be able
...
-'yttcc' and 'cconly') across all supported target infrastructures.
+Since existing ytt overlays are retained verbatim, new files to support
+ClusterClass are introduced in a non-overlapping fashion as separate
+directories (name 'yttcc' and 'cconly') across all supported target
+infrastructures.
 
-yttcc/ : contains overlays required to translate legacy config values into
+*yttcc/* : contains overlays required to translate legacy config values into
 CCluster.yaml. They will be included in an upcoming version of the CLI which
 supports ClusterClass-based cluster creation.
```
[the remark this answers is not in the document]
Comment (vuil) [reply docs/cli/clusterclass/cluster-class-support.md:40]: Good point, will do.

File: docs/cli/tkgctl/management-cluster-using-clusterclass.md (hunk 1 above)
Comment (vuil) [reply docs/cli/tkgctl/management-cluster-using-clusterclass.md]: Updated PR to mention tkg (composite) and framework packages and their relationships. The term 'composite' as I have seen mentioned was referring to fact that it is used to compose and drive the installation of a collection of other related sub-packages 
Deferring to @anujc25 to clarify the terminology we have been using.

File: docs/cli/clusterclass/cluster-class-support.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vuil) [reply docs/cli/clusterclass/cluster-class-support.md:47]: Done

File: docs/cli/tkgctl/management-cluster-using-clusterclass.md [hunk 3]
```diff
@@ -1,20 +1,23 @@
 # Management Cluster Deployment using ClusterClass
 
-Below is the sequence of management-cluster creation using clusterclass.
+Below is the sequence of steps involved in creating a management cluster ClusterClass.
```
[the remark this answers is not in the document]
Comment (vuil) [reply docs/cli/tkgctl/management-cluster-using-clusterclass.md]: done

File: docs/cli/tkgctl/management-cluster-using-clusterclass.md (hunk 1 above)
Comment (vuil) [reply docs/cli/tkgctl/management-cluster-using-clusterclass.md]: Done

File: docs/cli/tkgctl/management-cluster-using-clusterclass.md [hunk 4]
```diff
@@ -1,20 +1,46 @@
...
 1. Wait for packages to get deployed on management-cluster
 1. Apply `CCluster.yaml` to create management-cluster using default clusterclass
 1. Wait for CAPx controllers to reconcile the CCluster and deploy a management-cluster on specified infra
 1. Deploy `kapp-controller` on the management-cluster
-1. Addson-controller running on bootstrap cluster will deploy addons-components like CNI, CPI, CSI etc on the management-cluster
-1. Deploy `providers` on management-cluster
-1. Deploy `TKG meta package` to management-cluster
-    * Hack: Using `_ADDITIONAL_MANAGEMENT_COMPONENT_CONFIGURATION_FILE` to deploy additional components that are not getting deployed as part of `TKG meta package` at the moment. This includes TKR package related resource. This is only for testing purpose and should be removed once TKG package related changes are in.
+1. Addon-controller running on bootstrap cluster will deploy addons-components like CNI, CPI, CSI etc on the management-cluster
+1. Deploy CAPI `providers` on management-cluster
```
Comment (dharmjit) [inline docs/cli/tkgctl/management-cluster-using-clusterclass.md:16]: Question: Should it be
```suggestion
1. Deploy `CAPI` providers along with `CAPx` infra provider on management-cluster
```

Comment (vuil) [reply docs/cli/tkgctl/management-cluster-using-clusterclass.md:16]: thanks, by CAPI providers I was including both CAPI and infra specific providers, but I like your suggestion for better claritfy. Fixed.

File: docs/cli/tkgctl/management-cluster-using-clusterclass.md [hunk 5]
```diff
@@ -1,20 +1,23 @@
 # Management Cluster Deployment using ClusterClass
 
-Below is the sequence of management-cluster creation using clusterclass.
+Below is the sequence of steps involved in creating a management cluster ClusterClass.
 
 1. Deploy a bootstrap kind cluster
 1. Deploy `kapp-controller` on bootstrap cluster
 1. Deploy `providers` on bootstrap cluster
-1. Deploy `TKG meta package` to bootstrap cluster
+1. Deploy `tanzu-framework package` to bootstrap cluster
```
[the remark this answers is not in the document]
Comment (vuil) [reply docs/cli/tkgctl/management-cluster-using-clusterclass.md]: Thanks, forgot to update this one.

**Review by anujc25 — APPROVED** [review]
LGTM

**Pull request merged**